### PR TITLE
Awilliams/job cancel deprecation

### PR DESCRIPTION
**Pull request by mrwillyees**
# Description

Async cancel handler to inform user when cancel has been completed
## Type of change

- [ ] Bug fix
- [ ] New sample implementation
- [ ] Sample viewer enhancement
- [x] Other enhancement

**Platforms tested on**:

- [ ] Windows
- [ ] Android
- [ ] Linux
- [x] macOS
- [ ] iOS

## Checklist

- [ ] Runs and compiles on all active platforms as a standalone sample
- [x] Runs and compiles in the sample viewer(s)
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [ ] Code is commented with correct formatting (CTRL+i)
- [x] All variable and method names are camel case
- [x] There is no leftover commented code
- [x] Screenshots are correct size and display in description tab (500px by 500px, platform agnostic)
- [x] If adding a new sample, it is added to the sample viewer
- [ ] Cherry-picked to Main branch (if applicable)

### CODE REVIEW

**Comment by Gela** [comment]
@mrwillyees PR is still in draft. Please mark as in review and let me know when it's ready

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/ExportVectorTiles/ExportVectorTiles.qml [hunk 1]
```diff
@@ -274,6 +284,28 @@ Rectangle {
...
+        }
+
+        Connections {
+            id: exportVectorTilesJobConnections
+            ignoreUnknownSignals: true
+            function onCancelAsyncTaskStatusChanged() {
+                if(exportVectorTilesTask.exportVectorTilesJob.cancelAsyncTaskStatus === Enums.TaskStatusCompleted)
+                    statusTextCanceled.text = "Job canceled successfully"
+                else if(exportVectorTilesTask.exportVectorTilesJob.cancelAsyncTaskStatus === Enums.TaskStatusCompleted)
+                    statusTextCanceled.text = "Job canceled not successfully"
```
Comment (Gela) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/ExportVectorTiles/ExportVectorTiles.qml:301]: I would update the error message to something like: "Job failed to cancel successfully"

**Review by Gela — APPROVED** [review]
One small suggestion. I am seeing a couple of issues in the C++ sample only. I don't think these are related to your changes as they are also present in v.next but can you open an issue in Qt common to investigate them further?

1. The spinning wheel does not show on C++ while exporting. 
2. If I cancel the export mid-progress, both the green and red outlines stay visible. Again, this is C++ only, QML works as expected.

export

export

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/ExportVectorTiles/ExportVectorTiles.qml (hunk 1 above)
Comment (mrwillyees) [reply ArcGISRuntimeSDKQt_QMLSamples/Layers/ExportVectorTiles/ExportVectorTiles.qml:301]: I have noticed the not spinning busy indicator and put a fix in, surprised to see it not working again!

**Comment by mrwillyees** [comment]
I have noticed the not spinning busy indicator and put a fix in, surprised to see it not working again!

**Comment by Gela** [comment]
I added a note about the busy indicator in the issue you opened since that's still broken too. Your changes look good, thanks

**Review by Gela — APPROVED** [review]

**Pull request merged**